### ISSUE TITLE
Procedure to create custom HLG HDR curve for DirecTV

### ISSUE THREAD

**Issue by rtguy** [arvehj/jvcprojectortools#19]
Hi - Sorry this is more of a question than an "issue", however I did not see an option to post questions. Can you please explain the procedure to create a custom HLG HDR curve that could work well for large screens that are lacking in brightness?

I am familiar with this process for creating a custom curve for this purpose for HDR10. Would I just follow the same exact procedure, except starting with a HLG preset curve instead of a HDR10 preset?

For instance the readme says:

> Load preset gamma curve
> Select a preset gamma curve.
> 
> hdr pq - HDR gamma curve for content up to 4000nits
> hdr pq 1200 - HDR gamma curve for content up to 1200nits

Would I use something like "hdr hlg" for a HLG curve for content up to 4000nits as a starting point, then make my aggressive customizations to try and make it work better for a dim screen?

Or following this:

> eotf
> Select electro-optical transfer function to use. To select the most common hdr function enter "eo eotf_pq". Enter "eo" to get a menu of all supported functions.
> 
> eotf_bt1886 - SDR eotf with black level compensation
> eotf_hlg - HDR Hybrid Log-Gamma eotf defined by ITU-R BT.2100-0
> eotf_pq - HDR Perceptual Quantization eotf defined by ITU-R BT.2100-0
> eotf_gamma_2_2 - SDR gamma 2.2
> eotf_gamma_2_4 - SDR gamma 2.4

Would I just start with a blank slate and do "eo eotf_hlg"?

If you could please provide a high level overview of what commands I'd enter, once I start the tool, to get into a position to have a good baseline HLG curve, which I would then use things like bw and such commands to customize the curve, that would be really helpful. Thank you very much.

**Comment by arvehj** [comment]
Yes, you can use the same procedure a for HDR10. HLG is max 1000nits though. If you have a pq curve for content near that, you can start with that curve and change the eotf. There is also a hlg preset, but I have not tested this as I have not seen any hlg content.

**Comment by rtguy** [comment]
Thanks. What would be the process to getting a baseline HLG curve to then tweak, starting from scratch? Would I load a preset gamma like issue the command "hdr hlg" (if that is even a valid command)? Or instead use "eo eotf_hlg" and tweak from there?

If you are interested in seeing some sample HLG content, I believe these demo clips have it:
http://4kmedia.org/lg-cymatic-jazz-hdr-hlg-uhd-4k-demo/
http://4kmedia.org/travelxp-4k-hdr-hlg-sample/

**Comment by arvehj** [comment]
Either one should work. The preset is hdr_hlg if you want to use that

**Comment by rtguy** [comment]
Hi - I tried creating an HLG custom curve and wanted to get your thoughts please.

I simply ran the tool and used lp 4 to load the predefined HLG curve, then sent it as-is to the JVC. When playing the HLG demo content (listed in the message above) much of the highlights are over-blown (like as if the contrast was set way too high).

I am confused by this. I have a large screen which provides only about 50 nits, so I expected the image to look too dim, like it would for a standard HDR10 curve. Would that not be the case for HLG?

To tame the problem I used the Set With Contrast option and dialed it down. I also lowered sc and se. I then had a HLG curve that looked better than SDR with the same content without only mild overblown highlights.  Is that the right approach, or am I missing something here?

**Comment by arvehj** [comment]
The hdr_hlg preset is brighter than the hdr-pq presets as it has not been tuned. If you want a similar brightness run "bm 400" after lp.
